### PR TITLE
Add account deletion via AWS Step Function

### PR DESCRIPTION
**Pull request by mariogalic**
## What does this change?

This allows users to self-service their account deletion. For some straightforward users the deletion is fully automatic, while for others it is still manual process performed by Userhelp  (but now with fewer manual steps).

This PR has no complicated business logic and only consumes deletion logic performed by the following [AWS Step Function](https://aws.amazon.com/step-functions/):

image

Example of immediate visual identification of a failed step in the flow:

image

## What is the value of this and can you measure success?

Account deletion continues to be one of the most common Userhelp requests (at least 100 per week). Currently it is a manual and time consuming process involving at least 13 steps. This PR alleviates some of this pain. Please talk to Userhelp @andrewfindlay for details on this process.

A secondary aspect of this PR is to test out on a non-trivial user journey the capabilities of AWS Step Functions which is a new AWS service released in December 2016, and which has some promising implications with respect to reliability of business critical paths. 

### Account Deletion Step Function

Clients need a secret API key to consume the API Gateway that sits in-front of the Step Function.

### Identity API

There are 3 layers of security on top of the endpoint: 
* secret Dotcom client token, and
* secure session cookie, and 
* AWS API Gateway secret API key

### Dotcom
In dotcom:
* the user has to be signed in
* the user has to re-enter password in deletion form to confirm deletion
* there is anti-forgery protection on the deletion form

## Does this affect other platforms?

* New Identity API endpoint: https://github.com/guardian/identity/pull/615
* Step Function and its Lambdas: https://github.com/guardian/identity-account-deletion/pull/1

## Screenshots

The following form at `GET /delete` (accessible only to authenticated users) captures the canned response that Userhelp currently always mails to the user requesting deletion: 

image

After the user re-enters their password and submits the form, the Step Function is executed and the user is presented with one of the following two forms depending on if the deletion was performed automatically:

After automatic deletion:

image

After manual deletion:

image

### CODE REVIEW

**Review by TBonnin — COMMENTED** [review]
lgtm. I have not looked at the html template though.

File: identity/app/controllers/AccountDeletionController.scala
Comment (mario-galic) [inline identity/app/controllers/AccountDeletionController.scala]: Nice spot. I removed the flat map.

**Review by paulbrown1982 — APPROVED** [review]
👍 Looks awesome (once the build passes)

File: identity/app/views/profile/deletion/accountDeletion.scala.html
Comment (paperboyo) [inline identity/app/views/profile/deletion/accountDeletion.scala.html]: typo

**Comment by sndrs** [comment]
any chance we could merge this?

**Comment by mario-galic** [comment]
@sndrs I am waiting for an approval for dependent PR: https://github.com/guardian/identity/pull/615

**Pull request merged**
